File: LeetCodeCPlusPlus/py82.cpp
#include <iostream>
using namespace std;
// 删除链表中含有重复值的项；快慢指针
struct ListNode {
      int val;
      ListNode *next;
      ListNode(int x) : val(x), next(NULL) {}
 };
 
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *dummy = new ListNode(0);
        dummy->next = head;
        ListNode *slow = dummy, *fast = dummy->next;
        while (fast!=NULL) {
            if (fast->next != NULL && fast->next->val == fast->val) {
                int temp = fast->val;
                while (fast !=NULL && fast->val == temp)
                    fast = fast->next;
            }else {
                slow->next = fast;
                slow = slow->next;
                fast = fast->next;
            }
        }
        slow->next = fast;
        return dummy->next;
    }
};

File: LeetCodePython/py82.py
# 删除链表中含有重复值的所有项
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        p = head
        result = ListNode(0)
        q = result
        while p and p.next:
            same = False
            while p.next and p.val == p.next.val:
                same = True
                p.next = p.next.next
            if not same:
                q.next = p
                q = q.next
            p = p.next
       
        q.next = p if p else None
        return result.next